sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -3202,8 +3202,16 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
       AnalyzeTableStatement(tableName, partitionSpec, noScan = ctx.identifier != null)
     } else {
       checkPartitionSpec()
-      AnalyzeColumnStatement(
-        tableName, Option(visitIdentifierSeq(ctx.identifierSeq())), allColumns = false)
+      val colsOpt = Option(visitIdentifierSeq(ctx.identifierSeq()))
+      colsOpt.foreach { cols =>
+        val duplicates = cols.groupBy(identity).filter(_._2.size > 1).keys
+        if (duplicates.nonEmpty) {
+          throw new ParseException(
```

Comment:
shall we check it in `CheckAnalysis`? It's not a parser error really

Reply:
+1

Reply:
Hmm..the problem is `AnalyzeColumnCommand` of `core` module is inaccessible from `CheckAnalysis` of `catalyst` module...

Reply:
Oh...I see. we have customCheck rules to extend.